Traffic Ops unit tests: Traffic Ops unit tests: TestAddHealth sometimes fails

---
<!--
************ STOP!! ************
If this issue identifies a security vulnerability, DO NOT submit it! Instead, contact
the Apache Traffic Control Security Team at security@trafficcontrol.apache.org and follow the
guidelines at https://apache.org/security regarding vulnerability disclosure.

- For *SUPPORT QUESTIONS*, use the #traffic-control channel on the ASF slack (https://s.apache.org/tc-slack-request)
or the Traffic Control Users mailing list (send an email to users-subscribe@trafficcontrol.apache.org to subscribe).
- Before submitting, please **SEARCH GITHUB** for a similar issue or PR
    * https://github.com/apache/trafficcontrol/issues
    * https://github.com/apache/trafficcontrol/pulls
-->
<!-- Do not submit security vulnerabilities or support requests here - see above -->
## This Bug Report affects these Traffic Control components:
<!-- delete all those that don't apply -->
- Traffic Ops - tests

## Current behavior:
<!-- Describe how the bug happens -->
`github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/deliveryservice.TestAddHealth` usually passes but sometimes fails:
```go
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
```

## Expected behavior:
<!-- Describe what the behavior would be without the bug -->
All unit tests should always pass

## Steps to reproduce:
<!-- If the current behavior is a bug, please provide the *STEPS TO REPRODUCE* and
include the applicable TC version.
-->
```shell
cd traffic_ops/traffic_ops_golang/deliveryservice
go test -c
failures=0
for ((i = 0; i < 100; i++)); do
  ./deliveryservice.test || (( ++failures ))
done;
echo "Failed ${failures} times out of ${i}"
```

<details><summary>Example output (click to expand)</summary>summary>

```go
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
PASS
PASS
PASS
PASS
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
PASS
PASS
PASS
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
PASS
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestAddHealth (0.00s)
health_test.go:134: expected ds2-topology to have 1 online and 0 offline caches, but got 0 online and 0 offline instead
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
```
</details>

--- a/traffic_ops/traffic_ops_golang/deliveryservice/health.go
+++ b/traffic_ops/traffic_ops_golang/deliveryservice/health.go
@@ -124,7 +124,6 @@ func addHealth(ds tc.DeliveryServiceName, data map[tc.CacheGroupName]tc.HealthDa
 	var ok bool
 	var topology string
 	var cacheGroupNameMap = make(map[string]bool)
-	var skip bool
 
 	if deliveryService, ok = crConfig.DeliveryServices[string(ds)]; !ok {
 		return map[tc.CacheGroupName]tc.HealthDataCacheGroup{}, 0, 0, errors.New("delivery service not found in CRConfig")
@@ -142,6 +141,7 @@ func addHealth(ds tc.DeliveryServiceName, data map[tc.CacheGroupName]tc.HealthDa
 		}
 	}
 	for cacheName, avail := range crStates.Caches {
+		var skip bool
 		cache, ok := crConfig.ContentServers[string(cacheName)]
 		if !ok {
 			continue // TODO warn?

